File: mercadolivre/src/main/java/com/jornadadev/mercadolivre/config/ExistsId.java
package com.jornadadev.mercadolivre.config;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Documented
@Constraint(validatedBy = ExistsIdValidator.class)
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface ExistsId {
    String message() default "{idNaoEncontrado}";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

    String fieldName();

    Class<?> domainClass();
}

File: yfood/src/main/kotlin/com/jornadadev/yfood/config/ExistsId.kt
package com.jornadadev.yfood.config

import java.lang.annotation.Documented
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import javax.validation.Constraint
import kotlin.reflect.KClass

@Documented
@Constraint(validatedBy = [ExistsIdValidator::class])
@Target(AnnotationTarget.FIELD)
@Retention(RetentionPolicy.RUNTIME)
annotation class ExistsId(
        val message: String = "{idNaoEncontrado}",
        val groups: Array<KClass<*>> = [],
        val payload: Array<KClass<Any>> = [],
        val fieldName: String,
        val domainClass: KClass<*>
)